### DISCUSSION TITLE
Is the GitHub app needed after the initial setup?

### DISCUSSION THREAD

**Discussion by jschuur** [Thinkmill/keystatic#102] [General]
I see that there is a [disclaimer](https://thinkmill.notion.site/Step-by-step-guide-76db9947899345ab8112deddf183d4aa) about the GitHub app just being a way to easily get started, so once my repo/Vercel project is set up, do I still need the GitHub app around?

Or the Vercel integration?

From what I can tell, the deployed dynamic /app portion of Keystatic is the CMS UI, it stores content in git via GitHub and triggers a rebuild to static assets for the pages via a Vercel build (via the usual git push deploy).

Are either the GitHub app or the Vercel integration involved in the regular build/publishing process?

**Comment by jschuur** [comment]
I am intrigued by the 'Integrates with Astro' part BTW. Looking at you, @flexdinesh! ;)

**Reply by simonswiss** [reply]
Take a look at this guide - this doesn't cover the Markdoc stuff @flexdinesh has been working on, but this should be useful to you (particularly since you showed interest in the `local` storage strategy instead of GitHub):

https://thinkmill.notion.site/Add-Keystatic-to-an-Astro-Project-312e59b9be1f43d480d3f77638da2910

**Comment by jschuur** [comment]
Particularly relevant question, because when I went to do a new install for just the blank template (after I initially set up the blog template), it'll try and add the GitHub app again under the same name (jschuur's Keystatic), which of course you can't do, so I had to rename it.

**Reply by simonswiss** [reply]
Yep, this is something we're trying to improve.

I'll write a guide about the manual setup, so folks don't need to rely on our "automatic starter with template" solution we put together to help folks give Keystatic a try.

**Comment by airtonix** [comment]
If this is anything like NetlifyCms (now called DecapCMS) then `yes`:

- it defines the OAuth login contract, that the clientside connection will use to authenticate and authorise your access to github repo in which your content lives (along with the actual website)
- while Vercel has preview environments, you should really have a different github app for local development than you do for your production/preview environments. To avoid this, NetlifyCms has `netlify-cms-proxy-server` which implements a valid git-backend and just writes content directly to your local disk. it's amazing for local development.
- this means that a KeyStatic site isn't truly static, it still needs to be deployed to an environment capable of server side compute, DecapCMS has the same restriction when deployed to providers other than netlify. You can see this as evidence of the oauth api endpoints here: https://github.com/Thinkmill/keystatic/blob/main/keystatic/pages/api/keystatic/%5B%5B...params%5D%5D.tsx. DecapCMS can do this because it leans on a Indentity service provided by Netlify that you don't need to worry about deploying, but it only works for projects deployed to the Netlify service.


I haven't explored KeyStatic fully, but if you want a workflow where there is no login, you only use the keystatic admin locally, then there'd need to be support for writing directly to local disk. DecapCMS supports this via pluggable storage backends.
 

not saying DecapCMS is great. There's definitely some issues with its UX, but they've had the benefit of being in the game for a long time.

It's good to see an alternative that does just waht DecapCMS does. 👍🏻
